app/logical/bulk_update_request_processor.rb

```diff
@@ -182,6 +198,16 @@ def process!
           tag = Tag.find_or_create_by_name(args[0])
           tag.update!(category: Tag.categories.value_for(args[1]))
 
+        when :deprecate
+          tag = Tag.find_or_create_by_name(args[0])
+          tag.update!(is_deprecated: true)
+          TagImplication.active.where(consequent_name: tag.name).each { |ti| ti.reject!(User.system) }
+          TagImplication.active.where(antecedent_name: tag.name).each { |ti| ti.reject!(User.system) }
```

Comment:
This could be done with a single line if you use an OR. It should be something like:
```ruby
TagImplication.active.where(consequent_name: tag.name).or(where(antecedent_name: tag.name)).each { |ti| ti.reject!(User.system) }
```

Reply:
This doesn't work for me, it fails with "undefined method `where'". I think it has to be `.or(TagImplication.active.where`, which becomes a really long line which is hard to read.

Reply:
Yeah, I know I had done it before myself, and I was just speaking off the cuff, since I don't currently have a up-to-date working instance. I was specifically looking at the following when coming up with that suggestion.

https://github.com/danbooru/danbooru/blob/036341d8bab735a94c7a8122b00e8adb489c84ec/app/models/tag_relationship.rb#L69-L71

Is it possible to do something like the following though?

```ruby
q = TagImplication.active
q.where(consequent_name: tag.name).or(q.where(antecedent_name: tag.name)).each { |ti| ti.reject!(User.system) }
```